src/handlers/query-user.ts

```diff
@@ -0,0 +1,70 @@
+import { Context } from "../types/context";
+
+async function checkUserAccess(context: Context, username: string) {
+  const { octokit, payload } = context;
+  if (!payload.organization || !payload.comment.user?.name) {
+    throw new Error("Missing Organization / User from payload, cannot check for organization membership.");
+  }
+  const { status } = await octokit.orgs.checkMembershipForUser({
+    username: payload.comment.user.name,
+    org: payload.organization.login,
+  });
+  // @ts-expect-error Somehow typing seems wrong but according to
+  // https://docs.github.com/en/rest/orgs/members?apiVersion=2022-11-28#check-organization-membership-for-a-user--status-codes
+  // 204 means the user is part of the Organization
+  if (status !== 204) {
+    await octokit.issues.createComment({
+      body: `\`\`\`User ${username} cannot request another user as it is not member of the organization.\`\`\``,
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      issue_number: payload.issue.number,
+    });
+    return false;
+  }
+  return true;
+}
+
+export async function queryUser(context: Context, username: string) {
+  const {
+    octokit,
+    payload,
+    adapters: { supabase },
+    config,
+  } = context;
+  const body: string[] = [];
+  try {
+    const {
+      data: { id },
+    } = await octokit.users.getByUsername({
+      username,
+    });
+    if (!config.allowPublicQuery && !(await checkUserAccess(context, username))) {
+      return;
+    }
+    const access = await supabase.access.getAccess(id);
+    const wallet = await supabase.wallet.getWallet(id);
+    if (!access && !wallet) {
+      body.push(`\`\`\`
+User information for ${username} was not found.
+\`\`\``);
+    } else {
+      body.push(`
+| Property | Value |
+|----------|-------|`);
+      if (wallet) {
+        body.push(`| Wallet | ${wallet.address} |`);
+      }
+      if (access) {
+        body.push(`| Access | \`\`\`${Array.isArray(access.labels) ? access.labels.join(", ") : JSON.stringify(access.labels, null, 2)}\`\`\` |`);
+      }
+    }
+    await octokit.issues.createComment({
+      body: body.join("\n"),
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      issue_number: payload.issue.number,
+    });
+  } catch (e) {
+    await context.logger.fatal(`Could not query user ${username}.`, e);
```

Comment:
Shouldn't the logger automatically post the comment? We should only use the logger then instead of octokit post comment.

Reply:
On `fatal` the logger does post the comment to the issue with the error and help output.